src/ngraph_deassign_clusters.cc

```diff
@@ -57,19 +57,62 @@ static void MaybeLogPlacement(const Graph* graph) {
   if (!config::IsLoggingPlacement()) return;
 
   std::map<int, std::set<const Node*>> final_cluster_map;
-
+  int number_of_nodes = 0, nodes_marked_for_clustering = 0,
+      nodes_assigned_cluster = 0;
   for (auto node : graph->nodes()) {
+    number_of_nodes++;
+    // Check marked for clustering
+    if (NodeIsMarkedForClustering(node)) {
+      nodes_marked_for_clustering++;
+    }
+
+    // Check Cluster Assignment
     int cluster_idx;
     if (!GetNodeCluster(node, &cluster_idx).ok()) {
       cluster_idx = -1;
+    } else {
+      nodes_assigned_cluster++;
     }
     final_cluster_map[cluster_idx].insert(node);
   }
 
+  int perc_marked_for_clustering_of_total =
+      number_of_nodes > 0 ? (int)(((float)nodes_marked_for_clustering) * 100.0 /
+                                  number_of_nodes)
+                          : 0;
```

Comment:
is `number_of_nodes` ever expected to be zero?
maybe we should fail there due to zero-division so that we would be aware of such a case

Reply:
might even:
`int number_of_nodes = graph->nodes().size() //assuming size exists`
`if (number_of_nodes<=0) throw error`

then proceed to sure-to-pass division

but this is a minor thing how we calculate `number_of_nodes`. all I am saying is my assumption is `number_of_nodes` is a natural number. so safeguarding division is not required.

Reply:
(int)(((float)nodes_marked_for_clustering) * 100.0 / number_of_nodes)

the (int) cast at the very top: is it required?

might as well make `perc_marked_for_clustering_of_total` float.

the (float) cast on `nodes_marked_for_clustering` is not required, but does not hurt our cause.

similar comments for all perf variables, they are essentially floats. `int`-ing them is a rounding off to nearest integer

Reply:
Wait, now that I think of it, maybe u r converting them to int, so that `cout` does not show a gazillion numbers after the decimal. In which case I am ok with treating percentage as an integer. (...maybe... setprecision (but it is persistent, so might have to unset the precision of cout, if u set it) or some-such. so int is fine by me, as it is more hassle-free)

Reply:
Yes, I am converting to int to round off to nearest integer. It is just to get an estimate of how many nodes got placed on nGraph, so we do not really care about the decimal point.

Regarding the number_of_nodes, we should not have number_of_nodes<=0 . I did the check for consistency.